src/common/components/game/GameArea.js

```diff
@@ -0,0 +1,282 @@
+      isMyTurn, isSpectator, isTutorial, message, orangePieces, player, playingCardType,
+      selectedTile, sfxQueue, socket, status, target, tutorialStep, usernames, winner,
+      onActivateObject, onClickEndGame, onClickGameArea, onForfeit, onNextTutorialStep,
+      onPassTurn, onPrevTutorialStep, onSelectTile, onSendChatMessage, onTutorialStep
+    } = this.props;
+    const { areaHeight, boardSize, chatOpen } = this.state;
+
+    if (message) {
+      return <FullscreenMessage message={message} height={areaHeight} background={this.loadBackground()} />;
+    }
+
+    return (
+      <div
+        id="gameArea"
+        className="gameArea"
+        ref={(gameArea) => { this.gameArea = gameArea; }}
+        style={
+          screenfull.isFullscreen ? {width: '100%', height: '100%'} : {}
+        }
+      >
+        <div>
+          <GameNotification text="It's your turn!" enabled={currentTurn === player} />
+          <Sfx queue={sfxQueue} />
+        </div>
+
+        <Paper
+          className="background"
+          style={{
+            position: 'relative',
+            marginRight: chatOpen ? 256 : 64,
+            height: screenfull.isFullscreen ? areaHeight + 64 : areaHeight,
+            background: `url(${this.loadBackground()})`
+          }}
+          onClick={onClickGameArea}
+        >
+          <div
+            className="background"
+            style={{
+              display: 'flex',
+              height: '100%',
+              alignItems: 'center',
+              justifyContent: 'space-between'
+            }}
+          >
+            <div
+              className="background"
+              style={{
+                display: 'flex',
+                flexDirection: 'column',
+                justifyContent: 'center',
+                alignItems: 'center',
+                marginLeft: 20,
+                zIndex: 9999
+              }}
+            >
+              <Timer
+                player={player}
+                currentTurn={currentTurn}
+                gameOver={gameOver}
+                isTutorial={isTutorial}
+                isMyTurn={isMyTurn}
+                isAttackHappening={isAttackHappening}
+                onPassTurn={onPassTurn} />
+              <div style={{
+                display: 'flex',
+                justifyContent: 'space-around',
+                marginTop: 10
+              }}>
+                <SoundToggle />
+                <FullscreenToggle onClick={this.handleToggleFullScreen} />
+              </div>
+            </div>
+            <div
+              className="background"
+              style={{
+                display: 'flex',
+                justifyContent: 'center',
+                marginRight: 20,
+                zIndex: 9999
+              }}
+            >
+              <TutorialTooltip
+                tutorialStep={tutorialStep}
+                enabled={tutorialStep && tutorialStep.tooltip.location === 'endTurnButton'}
```

Comment:
I don't know enough about the system yet, so feel free to ignore it if I'm wrong, but this bit of logic _feels_ like it belongs somewhere else. Don't know quite where, though.

Reply:
I can factor it out.